ProcessMaker/processmaker
# Implement Screen Template Sync Functionality from GitHub Repository

This PR introduces the synchronization functionality to import Screen Templates from the [screen templates](https://github.com/ProcessMaker/screen-templates) GitHub repository into a ProcessMaker instance.

## Changes Made

- Created a migration to drop the foreign key 'user_id' and update screen template columns:
    - Updated `user_id` to be nullable, allowing proper syncing with default screen templates that do not have a user set
    - Added a new `media_collection` column
    - Added a new `unique_template_id` column
- Implemented a sync command `php artisan processmaker:sync-screen-templates`
- Created a `SyncScreenTemplate` job for executing the synchronization process.

## How to Test

1. Run `php artisan migrate` in your local environment to apply database migrations.
2. Execute `php artisan processmaker:sync-screen-templates` to sync screen templates.
3. Verify in the database's `screen_templates` table that the `Test Screen Template` was successfully synced.
4. Navigate to Designer > Screen > My Templates and Public Templates in the UI.
5. Confirm that the template displays in both tabs.

**Note: Filtering of My Templates and Public Templates is not yet implemented.**

## Related Tickets & Packages

**SubTasks:**
[FOUR-13618](https://processmaker.atlassian.net/browse/FOUR-13618)
[FOUR-13619](https://processmaker.atlassian.net/browse/FOUR-13619)
[FOUR-13620](https://processmaker.atlassian.net/browse/FOUR-13620)
[FOUR-13621](https://processmaker.atlassian.net/browse/FOUR-13621)

ci:next
ci:OPEN_AI_NLQ_TO_PMQL_ENABLED=true
ci: OPEN_AI_PROCESS_TRANSLATIONS_ENABLED=true

## Code Review Checklist


[FOUR-13618]: https://processmaker.atlassian.net/browse/FOUR-13618?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
